furusiya/model/entities/party/player.py

```diff
@@ -36,17 +36,37 @@ def __init__(self):
 
         print("You hold your wicked-looking {} at the ready!".format(weapon_name))
 
+    def move_or_attack(self, dx, dy):
+        # TODO: Should this be part of the Fighter component?
+        # the coordinates the player is moving to/attacking
+        x = self.x + dx
+        y = self.y + dy
+
+        # try to find an attackable object there
+        for obj in Game.area_map.get_entities_on(x, y):
+            if obj.has_component(Fighter):
+                target = obj
+                break
+        else:
+            self.move(dx, dy)
+            Game.renderer.recompute_fov = True
+            return
+
+        self.get_component(Fighter).attack(target)
```

Comment:
Should this be under the `for` part?

Reply:
Not really? I don't see the point of getting the Fighter component of the player instance *when we're checking for targets*.

Reply:
Clearly, I did not (and still do not) understand what this code does.